### DISCUSSION TITLE
What is a build.gradle identified as?

### DISCUSSION THREAD

**Discussion by Tajae1532** [openrewrite/rewrite#4763] [Q&A] [answered]
Hello I'm working on a recipe that finds build.gradle and build.gradle.kts files and I want to know what are they under? I have tried sourceFile, sourcePath and PlainText but none of them pick it up.

**Comment by timtebeek** [comment, marked as answer]
hi @Tajae1532 ! I find it helps to look at existing recipes when exploring a new recipe. For `build.gradle` files look at this example:
https://github.com/openrewrite/rewrite/blob/1095923b6e433860af8f5d71d7ab9a2a8f0c8a06/rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeDependency.java#L157-L164

For .kts files look here:
- https://github.com/openrewrite/rewrite/issues/3291

**Reply by Tajae1532** [reply]
Hello, this is part of a greater recipe. What I mean is I want to independently find build.gradle files 
An if check

**Reply by timtebeek** [reply]
It's not clear to me what you're asking; might it help to look at the FindGradleProject recipe also used above?
https://github.com/openrewrite/rewrite/blob/1095923b6e433860af8f5d71d7ab9a2a8f0c8a06/rewrite-gradle/src/main/java/org/openrewrite/gradle/search/FindGradleProject.java#L32

**Reply by Tajae1532** [reply]
For example if I wanted to look for Dockerfile we can say 
if (text.sourcePath.fileName.toString() == "Dockerfile")

or if I wanted to find a pom.xml we can say
if (tree.sourcePath.endsWith("pom.xml")

or even a Jenkinsfile we can say
if (sourceFile.sourcePath.fileName.toString() == "Jenkinsfile")

The question I'm asking is what is the equivalent for finding build.gradle files

**Reply by timtebeek** [reply]
The equivalent would the that `return Preconditions.check(new FindGradleProject(FindGradleProject.SearchCriteria.Marker).getVisitor(), ... the actual visitor you want to run...);`. I'll connect you with a colleague of yours for these patterns.

**Reply by shanman190** [reply]
As Tim was eluding to, understanding your context better would help.

As a for instance, for imperative recipes there is the helper visitor `IsBuildGradle` which just inspects the source path. If you're wanting to act as a precondition, `FindSourceFiles` can be a friend. Then lastly there is the `FindGradleProject` recipe which introspects deeper than just the source path itself by checking that the LST `SourceFile` also has a `GradleProject` marker.

As Tim said, looking at the implementations of various existing recipes can be helpful in finding these sort of features.
